kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanBuilderImpl.java

```diff
@@ -68,7 +69,11 @@ public ScanBuilder withFilter(Engine engine, Predicate predicate) {
 
   @Override
   public ScanBuilder withReadSchema(Engine engine, StructType readSchema) {
-    // TODO: validate the readSchema is a subset of the table schema
+    for(StructField field : readSchema.fields()) {
+      if (!snapshotSchema.fields().contains(field)) {
+        throw new IllegalArgumentException(String.format("Field '%s' is not a table field", field.getName()));
+      }
+    }
```

Comment:
We want to allow pruning nested columns as well, I think this won't work for that

Reply:
Please also add a test for this

Reply:
Also this needs case sensitive comparison.

We may add a utility method to check recursively whether the given `StructType` is a subset of another `StructType`. Check the Spark code.